File: app/src/main/java/com/neome/api/meta/base/dto/DefnFieldPickRole.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import org.jetbrains.annotations.Nullable;

import com.neome.api.meta.base.dto.DefnFieldEditable;
import com.neome.api.meta.base.dto.DefnStudioMapOfDtoOption;
import com.neome.api.meta.base.Types.EnumDefnThemePickVariant;
import com.neome.api.meta.base.Types.MetaIdField;
import com.neome.api.meta.base.Types.MetaIdRole;

@SuppressWarnings("unused")
public class DefnFieldPickRole extends DefnFieldEditable
{
  @Nullable
  public Boolean allowSystemRoles;

  @Nullable
  public DefnStudioMapOfDtoOption callerRoleMap;

  @Nullable
  public MetaIdField defaultRoleFieldId;

  @Nullable
  public MetaIdRole defaultRoleId;

  @Nullable
  public MetaIdRole[] excludeRoleIdSet;

  @Nullable
  public MetaIdRole[] filterRoleIdSet;

  @Nullable
  public DefnStudioMapOfDtoOption includeOptionMap;

  @Nullable
  public Long pageSize;

  @Nullable
  public EnumDefnThemePickVariant showAs;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnFieldPickRole.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumDefnThemePickVariant
import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdRole

interface DefnFieldPickRole : DefnFieldEditable {
    val allowSystemRoles: Boolean?
    val callerRoleMap: DefnStudioMapOfDtoOption?
    val defaultRoleFieldId: MetaIdField?
    val defaultRoleId: MetaIdRole?
    val excludeRoleIdSet: List<MetaIdRole>?
    val filterRoleIdSet: List<MetaIdRole>?
    val includeOptionMap: DefnStudioMapOfDtoOption?
    val pageSize: Long?
    val showAs: EnumDefnThemePickVariant?
}
